week-2-wide-perspective/w02e02-data-manipulation/src/orders/tiles/TotalIncomeTile.ts

```diff
@@ -0,0 +1,12 @@
+import { Order } from '../order';
+
+export function setTotalIncomeTiles(orders: Order[]): void {
+    const componentId = 'total-income'
+    const mountPoint = document.querySelector(`[data-tile="${componentId}"]`)
+    const content = mountPoint?.querySelector('[data-content]')
+    
+    // Ta wartość powinna być wykalkulowana na podstawie kolekcji ordersFakeData
+    if(content) {
+        content.innerHTML = String(orders.reduce((a,b) => a + b.sale, 0).toFixed(2))
```

Comment:
Na przykładzie tego kawałka kodu @Dam1anM - chciałbym zasiać w Tobie inną koncepcję rozwiązania problemu.
zobacz tutaj liczysz sobie _total income_ i to jest jakaś wartość którą chcemy pokazać dla _EndUser_. No i spoko cool, to jest to. 
Jednak zauważ kto to liczy... - noo komponent. No tak, komponent. A kto powinien decydować jaki jest `totalIncome` ?
- gdzie jest nasze źródło prawdy na temat danych... ?
- co jeśli kolejny komponent też będzie musiał liczyć to samo (bo w jakimś punkcie będzie to potrzebne)

Co Ty na to, żeby to źródło prawdy było jedno w klasie: `class Orders { }` !

To tam mamy strumień danych i możemy nim manipulować, zobacz:
```typescript
class Orders {
//...
private _income = this._source.pipe(map((orders) => orders.reduce((a,b) => a + b.sale, 0).toFixed(2)))

totalIncome(): Observable<number> {
   return this._income;
}

}
```
> To można zapisać jeszcze bardziej `rxjs`'owo ale nie wiem na ile znasz `RxJS` więc nie chce "odlecieć" ale generalnie `orders` można przerobić na zimy strumień i wykorzystać piękno composition API (coś jak z zadania `w02e01`)

Jeśli połączymy to z wstrzyknięciem nie danych `:orders Order[]` tylko tutaj mamy dostęp do `orders: Orders`

To możemy zrobić tak:
```typescript
if(content) {
   orders.totalIncome().subscribe(value => {
      content.innerHTML = String(value)
   })
}
```


Podobne rzeczy powinieneś zastosować do pozostałych komponentów i to moja propozycja [!]
_challenge_ 👈 
- [ ] Przebuduj  w ten sposób z zastosowaniem "Jednego źródła prawdy" na temat danych z `Order[]` !
dorzuć tam odpowiednie strumienie w zależności od potrzeb komponentów.
- [ ] Dodaj faktyczne zapytanie `AJAX` albo symulację do swojego rozwiązania - bo tutaj tego brakło :)

Reply:
@michaljabi ja się totalnie zgadzam z Twoimi komentarzami. Mówiąc wprost wywodzę się z Angulara i serwisy, rxjs, singletony itd to jem codziennie. Problem mój w tych zadaniach polega na tym, że trudno mi określić jak daleko mam iść by ten kod "wyśróbować", a na ile to jest zadanie "na wprowadzenie, na przetarcie jakichś koncepcji". Czasem zaczynam rozwijać jakieś rozwiązania problemów z zadań i je obcinam, bo się orientuje, że może nie o to w zadaniu chodzi żeby gdzieś kompleksowo odpływać :)

Prawdziwe mięso dla mnie to bardziej tematy związane z Nestem i ogólnie tymi "backendowymi klimatami" ale rozumiem, że kursanci przychodzą z różnym bagażem doświadczeń i na każdym poziomie zadania są potrzebne :)

P.S
Oczywiście wrzucę poprawki :)

Reply:
Hej @Dam1anM  jak z poprawkami, jeszcze zostawiam To otwarte, żebyś dał znaka.
Pozdrawiam.